Fix lint warnings/no-warning-comments

njmulsqb opened the pull request:
Fixed couple of warnings.

Fixes partially #608

psiinon commented:
In general we shouldnt just delete TODOs without actually adressing the things they were about .. but in this case I think its reasonable :)

thc202 commented:
From which linter is that?

njmulsqb commented:
> In general we shouldnt just delete TODOs without actually adressing the things they were about .. but in this case I think its reasonable :)

Actually the linter says that such comments should not exist in a production software, what should I do with rest of the todo comments that are causing warnings?

njmulsqb commented:
> From which linter is that?

Default linter of hud, I guess eslint. Got the warnings by _./gradlew npmLintAllHud_

thc202 commented:
Thank you. Yes, ideally we should fix them all, either address the todo or remove if no longer relevant.

thc202 commented on src/main/zapHomeFiles/hud/tools/history.js:
```diff
@@ -6,14 +6,13 @@
 
 const History = (function () {
 	// Constants
-	// todo: could probably switch this to a config file?
```
This and the following can be removed from all tools.

njmulsqb replied:
Ok I will

njmulsqb commented:
> Thank you. Yes, ideally we should fix them all, either address the todo or remove if no longer relevant.

How can I be sure that a particular todo is no longer relevant?

thc202 commented:
Thought question, it really depends of what the todo is about :) If not sure just ask.

njmulsqb commented:
> Working to remove ` // Todo: change this to a util function that reads in a config file (json/xml)` now from code

Removed all of its occurrences

thc202 commented:
Thank you, could you revert the changes done to the lock file and fix up all the commits?

njmulsqb commented:
I have reverted changes in package-lock.json file and fixed up last 2 commits.
Should package-lock.json be added to .gitignore to avoid pushing changes again? Running `npm i` updates this file

njmulsqb commented:
These are the pending no-warning-comments in HUD.

    // TODO change to try loading from localstorage (src\main\zapHomeFiles\hud\tools\commonAlerts.js)
    // Todo: change this to 'continue' and figure out / fix stopBreaking (src\main\zapHomeFiles\hud\tools\break.js)
    // FIXME: remove after  https://github.com/zaproxy/zap-hud/pull/620 (src\main\zapHomeFiles\hud\display.js)
    // Todo: is this too hacky? (src\main\zapHomeFiles\hud\target\inject.js)
    // TODO showComments section - put this code in a separate file and inject ? (src\main\zapHomeFiles\hud\target\inject.js)
    // TODO showEnable section - put this code in a separate file and inject ? (src\main\zapHomeFiles\hud\target\inject.js)

Give me the go-ahead to remove them and then we can merge this PR

thc202 commented:
Better leave those for now. The commits were not fixed up yet.

njmulsqb commented:
> Better leave those for now. The commits were not fixed up yet.

I have merged the 2 consequent commits, but the earlier ones are from Oct last year, after which many more commits have been made as visible via `git log --oneline` how can I merge non-consecutive commits?

thc202 commented:
The following should do it:
```
git fetch upstream
git rebase -i upstream/main
```
Set the last two commits to fixup and done.

njmulsqb commented:
> git rebase -i upstream/main

I have these commits,
pick 91fe5c6 no-warning-comments resolved
pick 62222b2 Some more warning comments removed
pick 319133f build(deps): bump vue-i18n from 8.26.5 to 8.26.7
pick 90427c5 build(deps-dev): bump eslint-plugin-no-unsanitized from 3.2.0 to 4.0.1
pick c5794ea fix: remove some TODO comments to solve lint warnings
The last two were already fixed up. Are you talking about first two?

thc202 commented:
Remove the lines containing `319133f` and `90427c5` and set `62222b2` and `c5794ea` to `fixup`.

njmulsqb commented:
I think we're done with all the no-warning-comments now except the 4 as mentioned previously.

thc202 commented:
Thank you!
